<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Carriers/ Enzymes/ Targets/ Transporters Polypeptide parsers — cett_poly_doc • dbparser</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Carriers/ Enzymes/ Targets/ Transporters Polypeptide parsers — cett_poly_doc" />
<meta property="og:description" content="Extract descriptions of identified polypeptide targets, enzymes, carriers,
 or transporters." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dbparser</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/dbparser.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Database_Saving.html">Saving data into Database</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ropensci/dbparser/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Carriers/ Enzymes/ Targets/ Transporters Polypeptide parsers</h1>
    <small class="dont-index">Source: <a href='https://github.com/ropensci/dbparser/blob/master/R/cett_polypeptide_general_infromation_parsers.R'><code>R/cett_polypeptide_general_infromation_parsers.R</code></a></small>
    <div class="hidden name"><code>cett_poly_doc.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Extract descriptions of identified polypeptide targets, enzymes, carriers,
 or transporters.</p>
    </div>

    <pre class="usage"><span class='fu'>carriers_polypeptides</span>(
  <span class='kw'>save_table</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>save_csv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>csv_path</span> <span class='kw'>=</span> <span class='st'>"."</span>,
  <span class='kw'>override_csv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>database_connection</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)

<span class='fu'>enzymes_polypeptides</span>(
  <span class='kw'>save_table</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>save_csv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>csv_path</span> <span class='kw'>=</span> <span class='st'>"."</span>,
  <span class='kw'>override_csv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>database_connection</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)

<span class='fu'>targets_polypeptides</span>(
  <span class='kw'>save_table</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>save_csv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>csv_path</span> <span class='kw'>=</span> <span class='st'>"."</span>,
  <span class='kw'>override_csv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>database_connection</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)

<span class='fu'>transporters_polypeptides</span>(
  <span class='kw'>save_table</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>save_csv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>csv_path</span> <span class='kw'>=</span> <span class='st'>"."</span>,
  <span class='kw'>override_csv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>database_connection</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>save_table</th>
      <td><p>boolean, save table in database if true.</p></td>
    </tr>
    <tr>
      <th>save_csv</th>
      <td><p>boolean, save csv version of parsed tibble if true</p></td>
    </tr>
    <tr>
      <th>csv_path</th>
      <td><p>location to save csv files into it, default is current
location, save_csv must be true</p></td>
    </tr>
    <tr>
      <th>override_csv</th>
      <td><p>override existing csv, if any, in case it is true in the
new parse operation</p></td>
    </tr>
    <tr>
      <th>database_connection</th>
      <td><p>DBI connection object that holds a connection to
user defined database. If <code>save_table</code> is enabled without providing
value for this function an error will be thrown.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a tibble with 20 variables:</p><dl'>
  <dt>id</dt><dd><p><a href='https://www.uniprot.org/'>Universal Protein Resource
  (UniProt) identifier</a></p></dd>
  <dt>source</dt><dd><p>Specifies whether the identified polypeptide ID is
  associated with any of the following UniProt knowledge bases:
  Swiss-Prot, which is manually annotated and reviewed, or TrEMBL,
  which is automatically annotated and not reviewed.</p></dd>
  <dt>name</dt><dd><p></p></dd>
  <dt>general_function</dt><dd><p>General summary of the physiological function of
  the polypeptide</p></dd>
  <dt>specific_function</dt><dd><p>A more specific description of the polypeptide’s
   physiological function within the cell.</p></dd>
  <dt>gene_name</dt><dd><p>The short name commonly associated with the associated
   gene. Eg. PTGS1.</p></dd>
  <dt>locus</dt><dd><p>The specific chromosomal location or position of the gene’s
   sequence on a chromosome.</p></dd>
  <dt>cellular_location</dt><dd><p>The cellular location of the polypeptide.</p></dd>
  <dt>transmembrane_regions</dt><dd><p>Areas of the polypeptide sequence that span
   a biological membrane.</p></dd>
  <dt>signal_regions</dt><dd><p>Location of any signal peptides within the
   polypeptide sequence.</p></dd>
  <dt>theoretical_pi</dt><dd><p>Theoretical isoelectric point.</p></dd>
  <dt>molecular_weight</dt><dd><p>The molecular weight of the polypeptide.</p></dd>
  <dt>chromosome_location</dt><dd><p>The chromosomal location of the polypeptide
  gene</p></dd>
  <dt>organism</dt><dd><p>The organism in which this polypeptide functions.</p></dd>
  <dt>organism_ncbi_taxonomy_id</dt><dd><p></p></dd>
  <dt>amino_acid_sequence</dt><dd><p>The amino acid sequence of the polypeptide</p></dd>
  <dt>amino_acid_format</dt><dd><p></p></dd>
  <dt>gene_sequence</dt><dd><p>The sequence of the associated gene.</p></dd>
  <dt>gene_format</dt><dd><p></p></dd>
  <dt>parent_key</dt><dd><p>carrier/ target/ enzyme/ transporter id</p></dd>

</dl>

    <h2 class="hasAnchor" id="read-drugbank-xml-db"><a class="anchor" href="#read-drugbank-xml-db"></a>read_drugbank_xml_db</h2>

    

<p><code><a href='read_drugbank_xml_db.html'>read_drugbank_xml_db</a></code> function must be called first before any
parser.</p>
<p>If <code><a href='read_drugbank_xml_db.html'>read_drugbank_xml_db</a></code> is called before for any reason, so
no need to call it again before calling this function.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other cett: 
<code><a href='cett_actions_doc.html'>cett_actions_doc</a></code>,
<code><a href='cett_doc.html'>cett_doc</a></code>,
<code><a href='cett_ex_identity_doc.html'>cett_ex_identity_doc</a></code>,
<code><a href='cett_go_doc.html'>cett_go_doc</a></code>,
<code><a href='cett_poly_pfms_doc.html'>cett_poly_pfms_doc</a></code>,
<code><a href='cett_poly_syn_doc.html'>cett_poly_syn_doc</a></code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'># the same parameters and usage will be applied for any parser</span>
<span class='co'># return only the parsed tibble</span>
<span class='fu'><a href='run_all_parsers.html'>run_all_parsers</a></span>()

<span class='co'># will throw an error, as database_connection is NULL</span>
<span class='fu'><a href='run_all_parsers.html'>run_all_parsers</a></span>(<span class='kw'>save_table</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='co'># save in database in SQLite in memory database and return parsed tibble</span>
<span class='no'>sqlite_con</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>DBI</span><span class='kw ns'>::</span><span class='fu'><a href='https://rdrr.io/pkg/DBI/man/dbConnect.html'>dbConnect</a></span>(<span class='kw pkg'>RSQLite</span><span class='kw ns'>::</span><span class='fu'><a href='https://rsqlite.r-dbi.org/reference/SQLite.html'>SQLite</a></span>())
<span class='fu'><a href='run_all_parsers.html'>run_all_parsers</a></span>(<span class='kw'>save_table</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>database_connection</span> <span class='kw'>=</span> <span class='no'>sqlite_con</span>)

<span class='co'># save parsed tibble as csv if it does not exist in current location,</span>
<span class='co'># and return parsed tibble.</span>
<span class='co'># if the csv exist before read it and return its data.</span>
<span class='fu'><a href='run_all_parsers.html'>run_all_parsers</a></span>(<span class='kw'>save_csv</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='co'># save in database, save parsed tibble as csv,</span>
<span class='co'># if it does not exist in current location and return parsed tibble.</span>
<span class='co'># if the csv exist before read it and return its data.</span>
<span class='fu'><a href='run_all_parsers.html'>run_all_parsers</a></span>(<span class='kw'>save_table</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>save_csv</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
<span class='kw'>database_connection</span> <span class='kw'>=</span> <span class='no'>sqlite_con</span>)

<span class='co'># save parsed tibble as csv if it does not exist in given location,</span>
<span class='co'># and return parsed tibble.</span>
<span class='co'># if the csv exist before read it and return its data.</span>
<span class='fu'><a href='run_all_parsers.html'>run_all_parsers</a></span>(<span class='kw'>save_csv</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>csv_path</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)

<span class='co'># save parsed tibble as csv if it does not exist in current location and</span>
<span class='co'># return parsed tibble.</span>
<span class='co'># if the csv exist override it and return it.</span>
<span class='fu'><a href='run_all_parsers.html'>run_all_parsers</a></span>(<span class='kw'>save_csv</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>csv_path</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>override</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='http://mohammedfcis.github.io'>Mohammed Ali</a>, <a href='https://www.linkedin.com/in/aliezzat1985/'>Ali Ezzat</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


